### PR TITLE
feat: finish compare table

### PR DESCRIPTION
**Pull request by Megrax**

### CODE REVIEW

**Comment by d-bytebase** [comment]
Is this ready for @LiuJi-Jim to review?

**Comment by Megrax** [comment]
> Is this ready for @LiuJi-Jim to review?

Sure. 
@LiuJi-Jim PTAL.

File: frontend/components/CompareTable.tsx [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+        <Tooltip
+          delayDuration={0}
+          content={`$${getDigit(
+            expectedCost,
+            2
+          )} is calculated based on utilization ${(
+            searchConfig.utilization * 100
+          ).toFixed(0)}% for ${searchConfig.leaseLength} year${
+            searchConfig.leaseLength > 1 ? "s" : ""
+          } lease`}
```
Comment (LiuJi-Jim) [inline frontend/components/CompareTable.tsx:198]: shall we extract the interpolated values and make it easier to read `content` ?

File: frontend/utils/table.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  }
+  return 1;
+};
+
+export const comparer: {
+  [key in SorterColumn]: (
+    a: dataSource,
+    b: dataSource,
+    isAscending: boolean
+  ) => number;
```
Comment (LiuJi-Jim) [inline frontend/utils/table.ts:101]: Better to extract this type

**Review by LiuJi-Jim — COMMENTED** [review]
LG overall, with some minor comments.

**Review by LiuJi-Jim — APPROVED** [review]
LGTM

**Pull request merged**